features/provider/admin/backend_apis/edit.feature

```diff
@@ -0,0 +1,29 @@
+@javascript
+Feature: Backend API edit page
+
+  Background:
+    Given a provider is logged in
+
+  Scenario: Edit an existing backend API
+    Given an admin is at a backend api edit page
+    Then they will be able to edit the name, description and base url
+    But will not be able to edit its system name
+
+  Scenario: Form validation
+    Given an admin is at a backend api edit page
+    Then they will not be able to update it without a name
+    And they will not be able to update it without a valid url
+
+  Scenario: Edit a backend API with duplicate fields
+    Given an admin is at a backend api edit page
+    Then they will be able to update it with an existing name and url
+
+  Scenario: Delete a backend API used by product
+    Given a backend api that is being used by a product
+    When an admin tries to delete the backend api from its edit page
+    Then it is not possible to delete the backend
+
+  Scenario: Delete a backend API not used by any product
+    Given a backend api that is not being used by any products
```

Comment:
This seems to be another step dup for `a backend`. Or does something more?

Reply:
No, I guess they're the same. I wanted to make it explicit that no product is using the backend, so maybe it should add an assertion to make sure of this. In that case they wouldn't be duplicates.

Reply:
I think at every place this step required the backend not to be used.

My request is to get all `Given a backend` step to use the same format. Other than that LGTM, now scenarios are much clearer.

Reply:
There are 4 step the creates a backend, we can split them so that `Given a backend` is always used first:
```ruby
Given "a backend( api)" do
Given "an admin is at a backend api edit page" do
Given "a backend api that {is} being used by a(ny) product(s)" do |used|
Given 'a backend that is unavailable' do
```

```cucumber
Given a backend api
And an admin is at the backend edit page

Given a backend api
And the backend is (not) used by a product

Given a backend api
And the backend api is unavailable
```
WDYT?

Bonus question: are backends used by default? The default product always has a backend but any further new products/backends are created independently.